Added new kb article grid-remove-group-indent (#2908)
telerik/blazor-docs: knowledge-base/grid-remove-group-indent.md

---
title: Removing Group Indent in Grid for Blazor
description: This article demonstrates how to remove the group indent in the Grid for Blazor by overriding the default theme styles.
type: how-to
page_title: How to Remove Group Indentation in Blazor Grid
slug: grid-kb-remove-group-indent
tags: grid, blazor, grouping, indentation
res_type: kb
ticketid: 1684110
---

## Description

> This knowledge base article also answers the following questions:

Reviewer: ```suggestion
This knowledge base article answers the following questions:
```
There are no other questions preceding this section.

- Is there a way to remove the group indent and adjust the spacing without compromising the Grid behavior?
- How can I customize the grouping appearance in a Blazor Grid?
- How to hide the group indent and group header icons in a Blazor Grid?

> - What is the best way to adjust the appearance of grouped headers in a Blazor Grid?

Reviewer: The last question is not clear enough as there are multiple ways to customize the headers. Let's be more specific here and list that the example shows how to hide the expand/collapse icon for the group rather than general customization of the group headers.

## Environment

<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Grid for Blazor</td>
		</tr>
	</tbody>
</table>

## Solution

> To remove the group indent in the [Grid for Blazor](slug:grid-overview) and adjust the appearance of grouped headers, you will need to override the default theme styles. This solution involves applying custom CSS styles to the Grid component. Run the example below and group the Grid by some of the columns to see the removed group indentation:

Reviewer: ```suggestion
To remove the group indent in the [Grid for Blazor](slug:grid-overview) and hide the expand/collapse icons of the groups, you will need to override the default theme styles. This solution involves applying custom CSS styles to the Grid component. Run the example below and group the Grid by some of the columns to see the removed group indentation:
```
See the above comment.

````RAZOR

> <TelerikGrid Data=@GridData
>              Groupable="true"
>              Pageable="true"
>              Height="400px"
>              Class="custom-grouping">
>     <GridColumns>
>         <GridColumn Field=@nameof(Employee.Name) Groupable="false" />
>         <GridColumn Field=@nameof(Employee.Team) Title="Team" />
>         <GridColumn Field=@nameof(Employee.IsOnLeave) Title="On Vacation" />
>     </GridColumns>
> </TelerikGrid>
> 
> <style>
>     .custom-grouping.k-grid .k-group-col {
>         width: 0;
>     }
> 
>     .custom-grouping.k-grid .k-grouping-row .k-icon {
>         display: none;
>     }
> </style>
> 
> @code {
>     private List<Employee>? GridData { get; set; }
> 
>     protected override void OnInitialized()
>     {
>         GridData = new List<Employee>();
>         var rand = new Random();
>         for (int i = 0; i < 15; i++)
>         {
>             GridData.Add(new Employee()
>                 {
>                     EmployeeId = i,
>                     Name = "Employee " + i.ToString(),
>                     Team = "Team " + i % 3,
>                     IsOnLeave = i % 2 == 0
>                 });
>         }
>     }
> 
>     public class Employee
>     {
>         public int EmployeeId { get; set; }
>         public string Name { get; set; } = string.Empty;
>         public string Team { get; set; } = string.Empty;
>         public bool IsOnLeave { get; set; }
>     }
> }

Reviewer 2: ```suggestion
<TelerikChart Width="100%" Height="100%">
    <ChartSeriesItems>
        @foreach (var series in GraphDataPoints)
        {
            <ChartSeries Field="@nameof(GraphDataPoint.Value)"
                         Type="ChartSeriesType.Bar"
                         Name="@series.Label"
                         Color="@series.Color"
                         Data="@([series])">
                <ChartSeriesStack Enabled="true" />
            </ChartSeries>
        }
    </ChartSeriesItems>
    <ChartLegend Position="ChartLegendPosition.Right" />
</TelerikChart>

@code {
    private List<GraphDataPoint> GraphDataPoints { get; set; } = [
        new GraphDataPoint
        {
            Label = "Early Settlement Candidate",
            Value = 1024,
            Color = "#D46663"
        },
    new GraphDataPoint
         {
            Label = "Needs Discovery to Strategize",
            Value = 980,
            Color = "#F89995"
        },
    new GraphDataPoint
        {
            Label = "Potential Dispositive Candidate",
            Value = 1006,
            Color = "#FFC7C7"
        },
    new GraphDataPoint
         {
            Label = "Potential Trial Candidate",
            Value = 1003,
            Color =  "#BCDCCF",
        },
    new GraphDataPoint
        {
            Label = "Settlement Candidate",
            Value = 987,
            Color = "#79C8AB"
        }
    ];

    public class GraphDataPoint
    {
        public required string Color { get; set; }
        public required int Value { get; set; }
        public required string Label { get; set; }
    }
}
```
Change the data collection property name to PascalCase and move the `GraphDataModel` last in the `code` block
